episodes/03-explore-data.Rmd

```diff
@@ -0,0 +1,278 @@
+
+It is important to see if all the variables (columns) have the data type that we require. For instance, a column might have numbers stored as characters, which would not allow us to make calculations with those numbers.
+
+```{r inspecting-data-str}
+str(gapminder) 
+
+```
+
+We can see that the `gapminder` object is a data.frame with `r nrow(gapminder)` observations (rows) and `r ncol(gapminder)` variables (columns). 
+
+In each line after a `$` sign, we see the name of each column, its type and first few values. 
+
+
+### First look at the dataset
+There are multiple ways to explore a data set. Here are just a few examples:
+
+
+```{r}
+head(gapminder) # see first 5  rows of the data set
+
+summary(gapminder) # gives basic statistical information about each column. Information format differes by data type.
+
+nrow(gapminder) # returns number of rows in a dataset
+
+ncol(gapminder) # returns number of columns in a dataset
+
+```
+
+### Dollar sign ($)
+
+When you're analyzing a data set, you often need to access its specific columns.
+
+One handy way to access a column is using it's name and a dollar sign `$`: 
+```{r subset-dollar-sign}
+country_vec <- gapminder$country  # Notation means: From dataset gapminder, give me column country. You can see that the column accessed in this way is just a vector of characters. 
+
+head(country_vec)
+
+```
+Note that the calling a column with a `$` sign will return a *vector*, it's not a data frame anymore.
+
+
+# [Data frame Manipulation with dplyr](https://datacarpentry.org/r-intro-geospatial/06-dplyr/index.html) 
+
+## Select
+Let's start manipulating the data. 
+
+First, we will adapt our data set, by keeping only the columns we're interested in, using the `select()` function from the `dplyr` package:
+
+```{r dplyr-select}
+year_country_gdp <- select(gapminder, year, country, gdpPercap) 
+
+head(year_country_gdp)
+
+```
+
+## Pipe
+Now, this is not the most common notation when working with `dplyr` package. `dplyr` offers an operator `%>%` called a pipe, which allows you build up very complicated commands in a readable way.
+
+
+In newer installation of `R` you can also find a notation `|>` . This pipe works in a similar way. The main difference is that you don't need to load any packages to have it available.
+
+
+The `select()` statement with pipe would look like that:
+
+```{r dplyr-pipe}
+
+year_country_gdp <- gapminder %>% 
+  select(year,country,gdpPercap)
+
+head(year_country_gdp)
+
+```
+
+First we define data set, then - with the use of pipe we pass it on to the `select()` function. This way we can chain multiple functions together, which we will be doing now. 
+
+## Filter
+
+We already know how to select only the needed columns. But now, we also want to filter the data set via certain conditions with `filter()` function. Instead of doing it in separate steps, we can do it all together. 
```

Comment:
```suggestion
We already know how to select only the needed columns. But now, we also want to filter the rows of our data set via certain conditions with `filter()` function. Instead of doing it in separate steps, we can do it all together. 
```